src/main/java/org/junit/internal/runners/JUnit38ClassRunner.java

```diff
@@ -170,6 +174,13 @@ public void sort(Sorter sorter) {
         }
     }
 
+    public void order(GenericOrdering ordering) throws InvalidOrderingException {
```

Comment:
Perhaps we should chat over VC, but I'll try to answer

 If `Sorter` doesn't extend `Ordering` then we have to write extra code to allow creation of an `Ordering` from a `Sorter` or `Comparator` (since `Ordering` is a very generic concept, it seems reasonable that one could have an `Ordering` that performs a sort; to do otherwise would require people to choose an API based on the current implementation). 

Also, as I said earlier, if we allow `order()` to take in an `Ordering` and that ordering does a sort, then we'll get worse performance than we would if the caller was smart enough to "know" that they should instead call `sort()` (really they shouldn't call either, and instead apply the `Ordering` or `Sorter` to the `Runner`).

Note that I also made `Orderable` extend `Sortable`. We could change that if we think it's an extra burden to people that write `Orderable` runners to have to also make them `Sortable`. If you want to get a hint as to the changes I would need to make if  `Orderable` did not extend `Sortable`, look at the second commit in this pull (the "unreachable code" would be reachable).